modules/javafx.base/src/test/java/test/com/sun/javafx/collections/ObservableListWrapperTest.java

```diff
@@ -0,0 +1,111 @@
+
+import com.sun.javafx.collections.ObservableListWrapper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ObservableListWrapperTest {
+
+    @Test(expected = NullPointerException.class)
+    public void testRemoveAll_Null() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>());
+        list.removeAll((Collection)null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testRetainAll_Null() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>());
+        list.retainAll((Collection)null);
+    }
+
+    @Test
+    public void testRemoveAll_Empty() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>());
+        assertFalse(list.removeAll(Collections.EMPTY_LIST));
+        assertFalse(list.removeAll(Collections.EMPTY_SET));
+    }
+
+    @Test
+    public void testRetainAll_Empty() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>());
+        assertFalse(list.retainAll(Collections.EMPTY_LIST));
+        assertFalse(list.retainAll(Collections.EMPTY_SET));
+    }
+
+    @Test
+    public void testRemoveAll_Args() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>(Arrays.asList(1, 2, 3)));
+        assertFalse(list.removeAll(0));
+        assertEquals(3, list.size());
+        assertTrue(list.removeAll(1));
+        assertEquals(2, list.size());
+        assertFalse(list.removeAll(1));
+        assertEquals(2, list.size());
+        assertTrue(list.removeAll(1, 2));
+        assertEquals(1, list.size());
+        assertTrue(list.removeAll(3));
+        assertEquals(0, list.size());
+        assertFalse(list.removeAll(Collections.EMPTY_SET));
+        assertFalse(list.removeAll(1));
+        assertFalse(list.removeAll(1, 2));
+    }
+
+    @Test
+    public void testRetainAll_Args() {
+        ObservableList<Integer> list = new ObservableListWrapper<>(new ArrayList<>(Arrays.asList(1, 2, 3)));
+        assertFalse(list.retainAll(0, 1, 2, 3));
+        assertEquals(3, list.size());
+        assertFalse(list.retainAll(1, 2, 3));
+        assertEquals(3, list.size());
+        assertTrue(list.retainAll(2, 3));
+        assertEquals(2, list.size());
+        assertTrue(list.retainAll(3));
+        assertEquals(1, list.size());
+        assertTrue(list.retainAll(1,2));
+        assertEquals(0, list.size());
+        assertFalse(list.retainAll(Collections.EMPTY_SET));
+        assertFalse(list.retainAll(2,3));
+        assertFalse(list.retainAll(3));
+
+        list = new ObservableListWrapper<>(new ArrayList<>(Arrays.asList(1, 2, 3)));
+        assertTrue(list.retainAll(Collections.EMPTY_SET));
+        assertEquals(0, list.size());
+    }
```

Comment:
Can you add a test for `retainAll` (in a separate test method) that removes multiple elements in a single call? Something like the following:

* set list to `(1, 2, 3, 4, 5)`
* remove `(2, 4)`
* verify that the list equals `(1, 3, 5)`
* remove `(1, 5)`
* verify that the list equals `(3)`

Similarly, a test for `retainAll` that removes multiple elements in a single call?

* set list to `(1, 2, 3, 4, 5)`
* retain `(1, 3, 5)`
* verify that the list equals `(1, 3, 5)`
* retain `(3)`
* verify that the list equals `(3)`

Reply:
That's a good suggestion.